subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/TestRunResult.kt

```diff
@@ -95,8 +142,23 @@ data class TestRunResult(
 
     fun notReportedCount(): Int = notReported.lostTests.size
 
-    private fun Iterable<SimpleRunTest>.lineByLine() = joinToString(separator = "\n") { it.name }
+}
 
+private fun Throwable?.composeWith(throwable: Throwable?): Throwable? {
+    return when {
+        this != null -> {
+            if (throwable != null) {
+                val message = "${message}. \n ${throwable.message}"
+                CompositeException(
+                    message,
+                    arrayOf(this, throwable)
+                )
+            } else {
+                this
+            }
+        }
+        else -> throwable
+    }
 }
```

Comment:
I'd prefer to avoid triple nesting.

```kotlin
private fun Throwable?.composeWith(throwable: Throwable?): Throwable? {
    if (this == null) return throwable
    if (throwable == null) return this

    val message = "${message}. \n ${throwable.message}"
    return CompositeException(
        message,
        arrayOf(this, throwable)
    )
}
```